### DISCUSSION TITLE
Newtonsoft JsonProperty support in v4

### DISCUSSION THREAD

**Discussion by jnazaren** [Azure/azure-cosmos-dotnet-v3#2691] [Q&A] [answered]
I have a follow up question to #2064 - my team is currently migrating from v2 to v3, and we have a lot of entity classes with fields tagged `[JsonProperty("propertyName")]`, where "propertName" is usually the camel case version of the field name. While we used the `CamelCasePropertyNamesContractResolver` in some cases previously, we now realize it's not necessarily a great fit in all cases since it also forces dictionary keys to be camel case, so we're thinking of switching to `CosmosPropertyNamingPolicy.Default` universally. That having been said however, we're wondering whether you'll still be honoring Newtsonsoft's JsonPropertyAttributes in v4, given that you're moving away from supporting Newtonsoft Json all together?

**Comment by j82w** [comment, marked as answer]
The v4 SDK is based on the [Azure.Core](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/Azure.Core) guideline. There will not be any newtonsoft support by default, but there should be a away to inject your own custom serializer that would allow you to use newtonsoft.
